lib/services/access_checker.go

```diff
@@ -319,30 +321,67 @@ func (a *accessChecker) CheckAccess(r AccessCheckable, state AccessState, matche
 // GetKubeResources returns the allowed and denied Kubernetes Resources configured
 // for a user.
 func (a *accessChecker) GetKubeResources(cluster types.KubeCluster) (allowed, denied []types.KubernetesResource) {
-	if len(a.info.AllowedResourceIDs) > 0 {
-		for _, r := range a.info.AllowedResourceIDs {
-			if r.Kind == types.KindKubePod && r.Name == cluster.GetName() && r.ClusterName == a.localCluster {
-				splitted := strings.SplitN(r.SubResourceName, "/", 3)
-				// This condition should never happen since SubResourceName is validated
-				// but it's better to validate it.
-				if len(splitted) != 2 {
-					continue
-				}
-				allowed = append(allowed,
-					types.KubernetesResource{
-						Kind:      r.Kind,
-						Namespace: splitted[0],
-						Name:      splitted[1],
-					},
-				)
+	if len(a.info.AllowedResourceIDs) == 0 {
+		return a.RoleSet.GetKubeResources(cluster)
+	}
+
+	rolesAllowed, rolesDenied := a.RoleSet.GetKubeResources(cluster)
+	// Allways append the denied resources from the roles. This is because
+	// the denied resources from the roles take precedence over the allowed
+	// resources from the certificate.
+	denied = append(denied, rolesDenied...)
```

Comment:
nit: avoid the copy?
```suggestion
	// Always return the denied resources from the roles. This is because
	// the denied resources from the roles take precedence over the allowed
	// resources from the certificate (and both could be a regex).
	denied = rolesDenied
```